src/fundus/scraping/crawler.py

```diff
@@ -178,123 +430,24 @@ def filter_warc_path_by_date(path: str) -> bool:
-                through Article.exception. If set to "raise", all errors encountered during extraction will
-                be raised. Defaults to "suppress".
-            only_complete (Union[bool, ExtractionFilter]): Set a callable satisfying the ExtractionFilter
-                protocol as an extraction filter or use a boolean. If False, all articles will be yielded,
-                if True, only those with all attributes extracted. Defaults to ExtractionFilter letting
-                through all articles with at least title, body, and publishing_date set.
-            url_filter (Optional[URLFilter]): A callable object satisfying the URLFilter protocol to skip
-                URLs before download. This filter applies on both requested and responded URL. Defaults to None.
-            only_unique (bool): If set to True, articles yielded will be unique on the responded URL.
-                Always returns the first encountered article. Defaults to True.
-
-        Returns:
-            Iterator[Article]: An iterator yielding objects of type Article.
-        """
+        warc_paths = tuple(self._get_warc_paths())
 
-        if max_articles == 0:
-            return
-
-        if max_articles is None:
-            max_articles = -1
-
-        def build_extraction_filter() -> Optional[ExtractionFilter]:
-            if isinstance(only_complete, bool):
-                return (
-                    None
-                    if only_complete is False
-                    else lambda extracted: not all(
-                        bool(v) if not isinstance(v, Exception) else False for _, v in extracted.items()
-                    )
-                )
-            else:
-                return only_complete
-
-        warc_paths = self._get_warc_paths(start, end)
-        response_cache: Set[str] = set()
-
-        article_task: Callable[[str], Iterator[Article]] = partial(
+        article_task = partial(
             self._fetch_articles,
-            publishers=self.publishers,
+            publishers=publishers,
             error_handling=error_handling,
-            extraction_filter=build_extraction_filter(),
+            extraction_filter=extraction_filter,
             url_filter=url_filter,
         )
 
         if self.processes == 0:
-            article_iter = self._single_crawl(warc_paths, article_task)
+            yield from self._single_crawl(warc_paths, article_task)
         else:
-            article_iter = self._parallel_crawl(warc_paths, article_task)
-
-        for article_idx, article in enumerate(article_iter, start=1):
-            if not only_unique or article.html.responded_url not in response_cache:
-                response_cache.add(article.html.responded_url)
-                yield article
-            if article_idx == max_articles:
-                break
+            yield from self._parallel_crawl(warc_paths, article_task)
```

Comment:
This crashes when running on gruenau using the following code. The problem does not occur on a single thread.
```
from fundus import PublisherCollection, CCNewsCrawler
from datetime import datetime

publisher = PublisherCollection.de
crawler = CCNewsCrawler(publisher, start=datetime(2024, 3, 1))
for article in crawler.crawl(max_articles=30, only_complete=False):
    print(article)
```

With the following Tracestack:
```
multiprocessing.pool.RemoteTraceback:
"""
Traceback (most recent call last):
  File "/usr/local/anaconda3-2023.03/lib/python3.10/multiprocessing/pool.py", line 125, in worker
    result = (True, func(*args, **kwds))
  File "/usr/local/anaconda3-2023.03/lib/python3.10/multiprocessing/pool.py", line 48, in mapstar
    return list(map(*args))
  File "/vol/fob-vol3/nebenf20/breidina/Arbeit/fundus/src/fundus/scraping/crawler.py", line 69, in __call__
    return self._deserialize()(*args, **kwargs)
  File "/vol/fob-vol3/nebenf20/breidina/Arbeit/fundus/src/fundus/scraping/crawler.py", line 66, in _deserialize
    return cast(Callable[_P, _T], dill.loads(self._serialized_target))
  File "/usr/local/anaconda3-2023.03/lib/python3.10/site-packages/dill/_dill.py", line 286, in loads
    return load(file, ignore, **kwds)
  File "/usr/local/anaconda3-2023.03/lib/python3.10/site-packages/dill/_dill.py", line 272, in load
    return Unpickler(file, ignore=ignore, **kwds).load()
  File "/usr/local/anaconda3-2023.03/lib/python3.10/site-packages/dill/_dill.py", line 419, in load
    obj = StockUnpickler.load(self)
  File "/usr/local/anaconda3-2023.03/lib/python3.10/site-packages/dill/_dill.py", line 568, in _create_type
    return typeobj(*args)
  File "/usr/local/anaconda3-2023.03/lib/python3.10/typing.py", line 348, in __init_subclass__
    raise TypeError("Cannot subclass special typing classes")
TypeError: Cannot subclass special typing classes
"""

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/vol/fob-vol3/nebenf20/breidina/Arbeit/fundus/test.py", line 7, in <module>
    for article in crawler.crawl(max_articles=30, only_complete=False):
  File "/vol/fob-vol3/nebenf20/breidina/Arbeit/fundus/src/fundus/scraping/crawler.py", line 203, in crawl
    for article in self._build_article_iterator(
  File "/vol/fob-vol3/nebenf20/breidina/Arbeit/fundus/src/fundus/scraping/crawler.py", line 453, in _build_article_iterator
    yield from self._parallel_crawl(warc_paths, article_task)
  File "/vol/fob-vol3/nebenf20/breidina/Arbeit/fundus/src/fundus/scraping/crawler.py", line 380, in _parallel_crawl
    yield from pool_queue_iter(pool.map_async(serialized_article_task, warc_paths), article_queue)
  File "/vol/fob-vol3/nebenf20/breidina/Arbeit/fundus/src/fundus/scraping/crawler.py", line 110, in pool_queue_iter
    handle.get(timeout=0.1)
  File "/usr/local/anaconda3-2023.03/lib/python3.10/multiprocessing/pool.py", line 774, in get
    raise self._value
TypeError: Cannot subclass special typing classes
```

Reply:
Ah sorry, to try both variants I forced the value of self.processes to be 0 and then it ran and worked fine using the portion of code:
```
article_iter = self._single_crawl(warc_paths, article_task)
 yield from self._single_crawl(warc_paths, article_task)
```
But without forcing it to run with self.processes = 0, it crashed with the above mentioned error message

Reply:
As @addie9800 found out this is caused when using `typing-extensions` versions `4.4` and `4.5`. To prevent this I'm gonna set `typing-extension` >= `4.6`.

Reply:
I did a fresh install from this PR's branch and had no crashes when following the tutorials.